app/src/helpers/FilterPicker.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback } from "react";
+import { DatePicker, PickerInput, PickerList, RangeDatePicker } from "@epam/promo";
+import { RangeDatePickerValue } from "@epam/uui-components";
+import { FilterConfig, IEditable } from "@epam/uui-core";
+
+export interface FilterPickerProps<TFilter extends Record<string, any>> extends IEditable<TFilter> {
+    filterConfig: FilterConfig<TFilter> | undefined;
+    isDropdownNeeded: boolean;
+}
+
+const FilterPickerImpl = <TFilter extends Record<string, any>>(props: FilterPickerProps<TFilter>) => {
+    const { value, filterConfig, onValueChange } = props;
+
+    const handleChange = useCallback((value: TFilter[keyof TFilter]) => {
+        onValueChange({ [filterConfig.field]: value } as TFilter);
+    }, [filterConfig.field, onValueChange]);
+    
+    const Picker = props.isDropdownNeeded ? PickerInput : PickerList as any;
```

Comment:
do we need this condition? We plan to use separate components for DynamicFilter. And if it component, not heler, move it from 'helpers' folder